src/core/reactor.cc

```diff
@@ -1129,14 +1129,17 @@ void cpu_stall_detector::maybe_report() {
 //
 // We can do it a cheaper if we don't report suppressed backtraces.
 void cpu_stall_detector::on_signal() {
-    if (reap_event_and_check_spuriousness()) {
-        return;
-    }
     auto tasks_processed = engine().tasks_processed();
     auto last_seen = _last_tasks_processed_seen.load(std::memory_order_relaxed);
     if (!last_seen) {
         return; // stall detector in not active
-    } else if (last_seen == tasks_processed) { // no task was processed - report
+    } else if (last_seen == tasks_processed) {
+        // we defer to the check of spuriousness to inside this unlikely condition
+        // since the check itself can be costly, involving a syscall (reactor::now())
+        if (is_spurious_signal()) {
```

Comment:
clock::now() doesn't involve a syscall (it's implemented in vdso), and we're in an expensive place already (a signal handler). So if it improves accuracy, we can restore it to the original location.

Reply:
Ah, I forgot we moved to CLOCK_THREAD_CPUTIME_ID. So no need to change anything here.